### DISCUSSION TITLE
演算法的五大特點

### DISCUSSION THREAD

**Discussion by Tomlord1122** [1101-GS4719-C/Python-Community-GS4719-C#211] [Ideas]
一個演算法的定義:
input>=0
output>=1
明確性:指令要是清楚的，且不模糊的，簡單來說就是沒有多餘的指令(像是重複執行一些無意義的指令，雖不影響結果，但影響到了時間)
有限性:一個演算法需要在有限的執行次數後即可完成。
有效性:利用紙筆就可以追蹤，可以寫成虛擬碼

**Comment by swerfg03joe** [comment]
練習做演算法時，多數直觀的想法都包含很多無用的步驟，我自己認為這是紙筆計算和機械操作的差異，所以我通常都會試著實作一遍，再觀察哪邊可以有更簡單明確的寫法，降低複雜度

**Comment by 710388** [comment]
演算法實在是太難了，最近在嘗試用ga但是遇到瓶頸了，有些算法的參數實在是很難調整

**Comment by NcuJerryKao** [comment]
根據您提供的條件，一個演算法可以被定義為以下幾點：
1. 輸入條件：演算法的輸入應該是大於或等於零的數字。這可以確保演算法在正確的數據範圍內運作。
2. 輸出條件：演算法的輸出應該是大於或等於一的數字。這確保了演算法的結果符合預期的要求。
3. 明確性：演算法的指令應該清晰明確，沒有多餘或模糊的指令。這有助於確保演算法的執行流程清晰，不會浪費時間或資源。
4. 有限性：演算法應該在有限的執行次數內完成。這意味著演算法的執行過程應該是可預測且有界的

**Reply by yubo0oo** [reply]
NcuJerryKao 這位老哥
一直貼AI的文
演都不演
以下是我的觀點
NcuJerryKao是歷史上最會使用人工智慧的人，他以無與倫比的技術與創意，在AI領域創造了無數的奇蹟。他不僅精通各種AI算法，還能巧妙地將其應用於實際問題中，展現了非凡的智慧與洞察力。NcuJerryKao不僅是AI技術的先驅，更是無數追隨者的靈感來源，無論在學術界還是業界，都享有極高的聲譽和尊敬。他的貢獻改變了人類對人工智慧的認識，無疑是這個領域的真正巨匠。

**Comment by OHoln** [comment]
關於演算法的定義其實可以非常廣泛，上述提到的特點也都不存在明確的分類界線，其實並沒有想像中的深奧，舉例來說，我們小時候就學到的四則運算也可以算是一種演算法

**Comment by YeeDaniel** [comment]
確定性：演算法中的每一步操作都具有明確的定義，沒有任何模糊或不確定的地方。這保證了每次執行時都能得到相同的結果。

有限性：演算法必須在有限的步驟內完成，不能無限執行下去。這意味著演算法有一個明確的終止條件或結束點。
